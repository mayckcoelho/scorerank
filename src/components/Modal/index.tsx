import React from 'react';
import ModalDefault from "react-native-modal";
import ModalDetail from "../ModalDetail";

import {
    Content,
    Indicator,
    CloseButton,
    CloseText,
    ImageContent,
    Description,
    SubText,
    Button,
    ButtonText
} from './styles';

interface Props {
    isVisible: boolean;
    image: string | undefined;
    title: string;
    description: string | undefined;
    sub: string | undefined;
    actionText: string;
    close(): void;
    action(): void;
}

const Modal: React.FC<Props> = ({ isVisible, image, title, description, sub, actionText, close, action }) => {
    return (
        <ModalDetail isVisible={isVisible} close={close}>
            <ImageContent source={{ uri: image }} resizeMode="contain" />
            <Description style={{ alignSelf: "flex-start" }}>{title}</Description>
            <SubText>{sub}</SubText>
            <Description>{description}</Description>
            <Button onPress={action}>
                <ButtonText>{actionText}</ButtonText>
            </Button>
        </ModalDetail>
    )
}

export default Modal;