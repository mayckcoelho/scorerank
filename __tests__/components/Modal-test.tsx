import React from 'react';
import Modal from '../../src/components/Modal';
import renderer from 'react-test-renderer';

describe('Modal should render correctly', () => {
    it('match snapshot', () => {
        const modal = renderer
            .create(<Modal
                isVisible={true}
                image=""
                title="test"
                description=""
                sub=""
                actionText=""
                close={() => { }}
                action={() => { }} />)
            .toJSON();
        expect(modal).toMatchSnapshot();
    });
})