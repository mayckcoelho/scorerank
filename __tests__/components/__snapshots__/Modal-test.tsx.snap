// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly match snapshot 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 0,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderTopLeftRadius": 15,
            "borderTopRightRadius": 15,
            "bottom": 0,
            "height": "50%",
            "justifyContent": "space-between",
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 0,
            "position": "absolute",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
              "backgroundColor": "#ccc",
              "borderRadius": 50,
              "height": 5,
              "marginTop": 5,
              "width": 50,
            },
          ]
        }
      />
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "",
          }
        }
        style={
          Array [
            Object {
              "alignSelf": "center",
              "height": 70,
              "width": 120,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "color": "#757575",
              "fontSize": 20,
              "marginBottom": 5,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 5,
            },
            Object {
              "alignSelf": "flex-start",
            },
          ]
        }
      >
        test
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "color": "#b2b2b2",
              "fontSize": 16,
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "color": "#757575",
              "fontSize": 20,
              "marginBottom": 5,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 5,
            },
          ]
        }
      >
        
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ca6a38",
            "borderRadius": 10,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 30,
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ccc",
            "borderRadius": 10,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
            ]
          }
        >
          Fechar
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
